### PR TITLE
adding ansible remediation to enable fips on rhel7

### PR DESCRIPTION
**Pull request by aghassemlouei**
#### Description:

- Adding in remediation/fix content to enable fips on rhel7 systems per DISA-STIG RHEL-07-021350

#### Rationale:

- No previous Ansible remediation/fix content existed

- Compliments fixes introduced in #1909 and subsequently #1880

### CODE REVIEW

**Comment by matejak** [comment]
ok to test

File: linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  # Install dracut-fips and dracut-fips-aesni packages
+
+  - name: Ensure dracut-fips and dracut-fips-aesni are installed
+    package:
+      name: '{{item}}'
+      state: present
+    with_items:
+    - dracut-fips
+    - dracut-fips-aesni
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml]: AES-NI is not guaranteed on all processors and environments, so this should be dropped.

File: linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  # Rebuild grub to include fips and uuid values
+
+  - name: Rebuild grub to include fips and uuid
+    shell: grub2-mkconfig -o /boot/grub2/grub.cfg
+    tags:
+    - fips_make_config_grub
+      @ANSIBLE_TAGS@
+  when: fips_state.stdout == "0"
+
+  # Note - A reboot will be required for these changes to take into full effect
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml]: Please remove some of the extraneous comments.

File: linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    - fips=1
+    - boot=UUID={{ grub_uuid.stdout }}
+    tags:
+    - fips_kernel_update
+      @ANSIBLE_TAGS@
+
+  # Rebuild grub to include fips and uuid values
+
+  - name: Rebuild grub to include fips and uuid
+    shell: grub2-mkconfig -o /boot/grub2/grub.cfg
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml]: Don't think this works for EFI systems

File: linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  - name:  Disable prelinking
+    lineinfile:
+      dest: /etc/sysconfig/prelink
+      regexp: '^#?PRELINKING'
+      line: 'PRELINKING=no'
+    when: prelink_installed.results|selectattr("yumstate","match", "installed")|list|length !=0
+    notify: undo existing prelinking
+    tags:
+    - fips_disable_prelink
+      @ANSIBLE_TAGS@
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml]: Prelinking should be disabled before fips is enabled.

File: linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
+# platform = multi_platform_rhel, Red Hat Enterprise Linux 7
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml]: This remediation only applies to `Red Hat Enterprise Linux 7`

**Comment by jan-cerny** [comment]
Please remove the file linux_os/guide/system/software/integrity/crypto/configure_crypto_policy/.DS_Store from the PR.

**Comment by dahaic** [comment]
No reply for 6 months. Closing.

**Pull request closed without merging**